src/node.rs

```diff
@@ -1109,6 +1119,15 @@ impl<S: ForkSource + std::fmt::Debug> InMemoryNode<S> {
         let tx_hash = l2_tx.hash();
         log::info!("");
         log::info!("Executing {}", format!("{:?}", tx_hash).bold());
+
+        {
+            let mut inner = self
+                .inner
+                .write()
+                .map_err(|e| format!("Failed to acquire write lock: {}", e))?;
+            inner.filters.notify_new_pending_transaction(l2_tx.hash());
```

Comment:
nit: This can be simplified to use the `tx_hash` variable
```suggestion
            inner.filters.notify_new_pending_transaction(tx_hash);
```